### PR TITLE
[Dashboard] Update aiohttp.ClientSession to only be created in async contexts

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Update all aiohttp.ClientSessions to be created in async contexts

De-dupe some extra ClientSessions that are created and share one across all dashboard modules

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -237,8 +237,8 @@ def get_packages(self):
         ],
         "default": [
             # If adding dependencies necessary to launch the dashboard api server,
-            # please add it to python/ray/dashboard/optional_deps.py as well.
```
Comment (aslonnie) [inline python/setup.py:240]: could you keep this `python/ray` line? `dashboard` has been moved.

Comment (alanwguo) [reply python/setup.py:240]: whoops, must've been a merge error.

File: python/requirements.txt [hunk 2]
```diff
@@ -48,7 +48,7 @@ scikit-image==0.21.0
 prometheus_client>=0.7.1
 pandas
 tensorboardX
-aiohttp>=3.7,<3.10.0
+aiohttp>=3.7,!=3.9.0
```
Comment (can-anyscale) [inline python/requirements.txt:51]: no need `!=3.9.0`, thankks

**Comment by aslonnie** [comment]
I can help you merge it if tests all pass.

**Comment by aslonnie** [comment]
thanks for fixing this.

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
I am merging this.

**Pull request merged**